stories/index.stories.tsx

```diff
@@ -87,6 +94,26 @@ stories.add('Basic usage', () => {
   );
 });
 
+stories.add('With custom icons', () => {
+  const icon = select('Icon (without value)', iconMap, iconMap.calendar);
+  const clearIcon = select('Clear icon (with value)', iconMap, iconMap.close);
+  const useCustomIcon = boolean('Custom icon', false);
+  const useCustomClearIcon = boolean('Custom clear icon', false);
+  const CustomIcon = (props: any) => <button {...props}>Select</button>;
+  const CustomClearIcon = (props: any) => <button {...props}>Reset</button>;
+  const x = useCustomIcon ? ((<CustomIcon />) as unknown) : icon;
+  const y = useCustomClearIcon ? ((<CustomClearIcon />) as unknown) : clearIcon;
```

Comment:
I don't understand these unknowns

Reply:
This is a workaround since TS can't really understand that `<Component/>` returns a `React.ReactElement`. I think I can replace it with a `ts-ignore` if you think that would be less worse.